src/OpenSage.Game/Utilities/BitArrayMatchFinder.cs

```diff
@@ -0,0 +1,67 @@
+﻿using System;
+using OpenSage.Logic.Object;
+using OpenSage.Mathematics;
+
+namespace OpenSage.Utilities;
+
+// TODO(Port): Check if the logic here matches SparseMatchFinder.h.
```

Comment:
just as a heads-up, I'm pretty sure I wrote most of this logic originally. I wrote a bunch of unit tests for ModelConditionState and it passed those but... things just never really looked right in game. I don't know that this code is actually correct, so it will be good to take a look when porting SparseMatchFinder.

I'm also glad you moved it, because now your name is in the commit history for it 😆